lib/promise-m3u8.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Created by elad.benedict on 8/23/2015.
+ */
+
+var m3u8Parser = require('m3u8');
+var fs = require('fs');
+var Q = require('q');
+
+module.exports.M3U = m3u8Parser.M3U;
+
+module.exports.parseM3U8 = function(manifestStringOrStream, options){
+
+    var stream;
+    if (manifestStringOrStream.pipe)
+    {
+        stream = manifestStringOrStream;
+    }
+    else
+    {
+        if (options && options.verbatim)
+        {
+            // Use value verbatim
+            var Readable = require('stream').Readable;
+            var s = new Readable();
+            s.push(manifestStringOrStream);
+            s.push(null); // Stream end
+            stream = s;
+        }
+        else
+        {
+            // Value is the M3U8 path
+            stream = fs.createReadStream(manifestStringOrStream);
```

Comment:
createReadStream can throw errors.

HLSUtils.readM3u8File = function(dest) {

        var deferred = Q.defer();
        var parser = m3u8Parser.createStream();
        var playlist = fs.createReadStream(dest);

        // This will wait until we know the readable stream is actually valid before piping
        playlist.on('open', function () {
            console.log('stream is open');
            playlist.pipe(parser);
        });

        // This catches any errors that happen while creating the readable stream (usually invalid names)
        playlist.on('error', function(err) {
            console.log('error: ' + err);
            deferred.reject(err);   //TODO reject parameter
        });

        parser.on('m3u', function (m3u) {
            deferred.resolve(m3u);
        });
        parser.on('error', function (err) {
            deferred.reject(err);   //TODO reject parameter
        });
        return deferred.promise;
    };